x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -335,14 +336,22 @@ export class FleetPlugin
     });
     licenseService.start(this.licensing$);
 
-    const fleetServerSetup = startFleetServerSetup();
-
     this.telemetryEventsSender.start(plugins.telemetry, core);
 
     return {
       fleetSetupCompleted: () =>
-        new Promise<void>((resolve) => {
-          Promise.all([fleetServerSetup]).finally(() => resolve());
+        new Promise<void>(async (resolve, reject) => {
+          try {
+            await startFleetServerSetup();
+            await setupFleet(
+              new SavedObjectsClient(core.savedObjects.createInternalRepository()),
+              core.elasticsearch.client.asInternalUser
+            );
+
+            resolve();
+          } catch (error) {
+            reject(error);
+          }
```

Comment:
I think we move this promise constructor outside the `fleetSetupCompleted` function so that it isn't called each time `fleetSetupCompleted` is called and instead just return the already-constructed Promise. Also what is calling `fleetSetupCompleted`? I don't think we should rely on this being called for setup to run.

Also more of a nit, but I don't think you need the Promise constructor and you can just use an async function directly:
```ts
const fleetSetup = (async () {
  await startFleetServerSetup();
  await setupFleet(...);
})();
```

Reply:
Fleet doesn't call `fleetSetupCompleted` on its own, but the `security_solution` plugin does rely on it to handle some of their dependent setup logic here:

https://github.com/elastic/kibana/blob/828a5790e4b16ad13bd5cf2367e2082a24ad3c21/x-pack/plugins/security_solution/server/plugin.ts#L368-L381

Agreed on restructuring this - it doesn't need a `Promise` constructor anymore.

Reply:
> Fleet doesn't call `fleetSetupCompleted` on its own, but the `security_solution` plugin does rely on it to handle some of their dependent setup logic here

I still don't think we should rely on security solution to call this. In fact, we or they may remove this in 8.0 as part of removing our BWC code for pre-GA fleet. We can still support this API for now by initiating the Fleet setup directly in the `start` function before the return statement, but just don't await the promise and instead return it from `fleetSetupCompleted()`.

Reply:
Moved this logic out of `fleetSetupCompleted` so we don't need to rely on security solution calling it.